opt_einsum/paths.py

```diff
@@ -84,6 +81,56 @@ def optimal(input_sets, output_set, idx_dict, memory_limit):
     return path
 
 
+def _parse_possible_contraction(positions, input_sets, output_set, idx_dict,
+                                memory_limit, path_cost, naive_cost):
```

Comment:
Can you add a snippet about what this internal function does?

Reply:
Yes, will do.